extensions/socket/internal.m

```diff
@@ -0,0 +1,657 @@
+#import <LuaSkin/LuaSkin.h>
+#import <CocoaAsyncSocket/GCDAsyncSocket.h>
+#import "socket.h"
+
+// Definitions
+@interface HSAsyncSocket : GCDAsyncSocket
+@property int readCallback;
+@property int writeCallback;
+@property int connectCallback;
+@property NSTimeInterval timeout;
+@property NSMutableArray* connectedSockets;
+@end
+
+// Userdata for hs.socket objects
+static const char *USERDATA_TAG = "hs.socket";
+#define getUserData(L, idx) (__bridge HSAsyncSocket *)((asyncSocketUserData *)lua_touserdata(L, idx))->asyncSocket;
+
+
+// Delegate implementation
+@implementation HSAsyncSocket
+
+- (id)init {
+    self.readCallback = LUA_NOREF;
+    self.writeCallback = LUA_NOREF;
+    self.connectCallback = LUA_NOREF;
+    self.timeout = -1;
+    self.connectedSockets = [[NSMutableArray alloc] init];
+    return [super initWithDelegate:self delegateQueue:dispatch_get_main_queue()];
+}
+
+- (void)socket:(HSAsyncSocket *)sock didConnectToHost:(NSString *)host port:(UInt16)port {
```

Comment:
How feasible/difficult would it be to have LuaSkin's log* methods automatically call `dispatch_async` to the main thread for log operations performed on background threads, with everything else being subjected to the `if (![NSThread isMainThread])` check performed in `+ (id)shared `? Seems to me like that wouldn't be terribly unsafe, but might be too much work for not a lot of benefit.

In any case I've converted it to use the background queue with `dispatch_async` to the main thread where needed and it passes all tests.